table_schema_cache.go

```diff
@@ -211,6 +206,81 @@ func LoadTables(db *sql.DB, tableFilter TableFilter, columnCompressionConfig Col
 	return tableSchemaCache, nil
 }
 
+func (t *TableSchema) findColumnByName(name string, err error) (*schema.TableColumn, error) {
+	for _, column := range t.Columns {
+		if column.Name == name {
+			return &column, nil
+		}
+	}
+	return nil, err
```

Comment:
At first I thought this `err` is always `nil` because no where in this function is it initialized. Then I see that you're passing the same `NonExistingPaginationKeyColumnError(dbname, table, name)` in as an `err` argument and returning it. 

We can just build that error in this function directly. dbname and table are all information available directly on the  TableSchema struct that this is a method of.

Reply:
Nice suggestion! That access would allow me to also simplify `paginationKeyColumn`-method even further